core-js/src/main/javascript/cdf/dashboard/RefreshEngine.js

```diff
@@ -370,16 +384,22 @@ define([], function() {
       fireRefresh: function() {
         activeTimer = null;
         var currentTime = getCurrentTime();
+        var component = null;
 
         while(refreshQueue.length > 0 && refreshQueue[0].nextRefresh <= currentTime) {
           var info = refreshQueue.shift();//pop first
-          //call update, which calls processComponent
-          refreshComponent(info.component);
+          component = info.component;
+
+          //call update, which calls processComponent, but only if not in the dashboard update queue
+          if(!dashboard.updateQueue || dashboard.updateQueue.indexOf(component) == -1) {
+            //setTimeout(function() {
+              refreshComponent(component);
+            //}, 10);
+            reinsertInQueue(component);
```

Comment:
Shouldn't this be in an `else` branch? refresh or reinsert? In that case, because it is not in the queue, we could simply insert in the queue again?

Reply:
I made some subtle changes to the way that the fireResfresh works, and created a reinsertInQueue, with the objective of making the calls to those methods (fireRefresh, and (clearFromQueue+insertInQueue)) non-destructive.

This way I guarantee that the component is in the refresh queue (he should be there anyway because it was explicitly added when the dashboard instance was created), even while the component lifecycle is progressing through it's phases (and hence, that if to not refresh the component if it is in the dashboard update queue).

This last part somehow bleeds some of the logic that should be part of the dashboard lifecycle itself (don't care if it is asked to update multiple times).... but the true is that the behavior that it is applying doesn't work well with the way the refresh engine was built.

**Notice**: the commented lines will be removed.... they were commited by error ( I shall fix that before)